urumanifest/github.py

```diff
@@ -396,19 +405,21 @@ def iter_client_dir():
             leave=False
         ) as progress:
             for info in progress:
-                logging.trace(f"Extracting {info.filename}")
-                _unpack_member(output_path, archive, info)
+                logging.trace(f"Extracting {info[1].filename}")
+                _unpack_member(output_path, info[0], archive, info[1])
 
         gather_key = workflow_lut[name]
-        gather_package = { gather_key: [PurePosixPath(i.filename).name for i in desired_members] }
+        gather_package = { gather_key: [i[2] for i in desired_members] }
         with output_path.joinpath("control.json").open("w") as fp:
             json.dump(gather_package, fp, indent=2)
 
         database.workflow_gathers[name] = subdir_name
 
-def _unpack_member(output_path: Path, archive: zipfile.ZipFile, info: zipfile.ZipInfo):
+def _unpack_member(output_path: Path, output_subpath: Path, archive: zipfile.ZipFile, info: zipfile.ZipInfo):
     with archive.open(info, "r") as infile:
-        with output_path.joinpath(Path(info.filename).name).open("wb") as outfile:
+        full_path = output_path.joinpath(output_subpath)
+        Path(os.path.dirname(full_path)).mkdir(parents=True, exist_ok=True)
```

Comment:
What is going on here? This looks like it's making a new directory in the CWD...

Reply:
It's unpacking the app bundle into the gather directory - but it needs to create any missing folders in the unpack path along the way.